test/EFCore.PG.FunctionalTests/TestUtilities/NpgsqlTestStore.cs

```diff
@@ -416,6 +416,26 @@ private static Task<IEnumerable<T>> QueryAsync<T>(DbConnection connection, strin
             return command;
         }
 
+        public Version GetPostgresVersion()
+        {
+            var opened = false;
+            if (Connection.State == ConnectionState.Closed)
+            {
+                Connection.Open();
+                opened = true;
+            }
+
+            try
+            {
+                return ((NpgsqlConnection)Connection).PostgreSqlVersion;
```

Comment:
Should we instead set this up with the EF Core configured options (e.g. `NpgsqlOptions`)? Then we could run these tests for one with "version" 10 and one with "version" 11 (configured versions, not real backend versions).

Reply:
I guess that depends on the question above, of running our tests on multiple backend versions.

If we do end up doing that, we'd definitely want to run on actual backend versions and not just produce SQL for the different versions (because technically SQL we produce in PostgreSQL 10 compatibility mode could pass on 11 but fail on 10). So we'd need both real backends for the different versions, and some way to apply the version on the context options of all tests (probably not trivial).

If we do it like that, then I don't think it really matters whether we test the real database version here or the context option, as they're supposed to be aligned (unless we mess things up...).

In any case, in the immediate term this is necessary in order to skip the test on PostgreSQL 10 (which is what Appveyor has) - any other solution would probably take significantly more time, and we're only talking about test infrastructure. So am definitely open to revisiting a bit later if you want.

Reply:
Makes sense and sounds good. The AppVeyor constraint was what I was overlooking.